app/controllers/api/v1/rooms_controller.rb

```diff
@@ -106,6 +106,23 @@ def find_room
       def room_params
         params.require(:room).permit(:name, :user_id, :presentation)
       end
+
+      # Generates an access code on rooms#create if the access code room configuration is enabled
+      def generate_access_code
+        access_code_settings = RoomSettingsGetter.new(
+          room_id: @new_room.id,
+          provider: current_provider,
+          current_user:,
+          show_codes: false,
+          settings: %w[glViewerAccessCode glModeratorAccessCode]
+        ).call
+
+        access_code_settings.each do |access_code_setting|
```

Comment:
I believe that the `access_code_settings` will always be `%w[glViewerAccessCode  glModeratorAccessCode]`,

Reply:
Besides, all points mentioned above.
Using `Hash#each_key` seems to be more adequate since not using the values,